### ISSUE TITLE
Hook After is used after first block of "context" and not after the last one

### ISSUE THREAD

**Issue by herobix** [cypress-io/cypress#1206]
- Operating System: Ubuntu 16-04
- Cypress Version: 1.4.1
- Browser Version: Chrome 63
### Is this a Feature or Bug?
Bug

### Current behavior:
Right under describe, I put in a **Before** and **After** hook. If I put in 2 different contexts with tests. 

When I run the test, **Before** is run in the beginning as designed. 
**After** runs after the FIRST context is run and not after all of the tests have been run. 

### Desired behavior:
When inserting **After** at the root level, it would make sense that it actives after all test after all tests have been run. 

Since my assumption is that it works like an if statement. So that my **After** would only execute when that specific describe finishes. 

Since the other **context** is a level above the **After** hook

### How to reproduce:
Run test code below 

#### Test code:

```js
describe('Testing "After" Hook', () => {

	before(() => {
		cy.log('this is before - root');
	});

	afterEach(() => {
		cy.log('this is afterEach - root');
	});

	after(() => {
		cy.log('This is After - root');
	});

	context(' - Part 01 - ', () => {

		it('Check test 01-01', () => {
			cy.log('Check test 01-01');
		});

		it('Check test 01-02', () => {
			cy.log('Check test 01-02');
		});
	});

	context(' - Part 02 - ', () => {
		beforeEach(() => {
			cy.log('this is beforeEach - Part 02');

		});

		it('Check test 02-01', () => {
			cy.log('Check test 02-01');
		});

		it('Check test 02-02', () => {
			cy.log('Check test 02-02');
		});
	});
});

```

### Additional Info (images, stack traces, etc)
![cypressafterhookissue-19jan2018](https://user-images.githubusercontent.com/17165035/35162432-33d3c874-fcf8-11e7-87ac-5554de79570d.png)

![cypressafterhookissue-console-19jan2018](https://user-images.githubusercontent.com/17165035/35162893-b6844888-fcf9-11e7-910e-ffa96eaedd2d.png)

**Comment by herobix** [comment]
Even moving **After** into the second **context** is still running **After** after the first **context** and not after all test have run

**Comment by jennifer-shehane** [comment]
Hi, I believe this is a duplicate of this issue #405 which we have been looking for reproducible code for, so this should be helpful!

TLDR: the `after` is executing in the correct place, but it is not displaying in the correct place in the Cypress Command Log.

**Comment by herobix** [comment]
I uploaded some more screenshots and it is like issue #405 
The weird part is that I have a cleanup script running to delete users (I know it is not the best way and all that jazz, it is what it is for now), with the setup, as shown above, **After** is actually not being triggered

**Comment by herobix** [comment]
Never mind, looks like it is triggering as it should
